app/src/features/pricing/components/PremiumFeature/index.tsx

```diff
@@ -0,0 +1,130 @@
+  features: FeatureLimitType[];
+  children?: React.ReactNode;
+  className?: string;
+  popoverPlacement: PopconfirmProps["placement"];
+  disabled?: boolean;
+  source: string;
+}
+
+export const PremiumFeature: React.FC<PremiumFeatureProps> = ({
+  onContinue,
+  children,
+  className = "",
+  features,
+  popoverPlacement,
+  disabled = false,
+  source,
+}) => {
+  const dispatch = useDispatch();
+  const user = useSelector(getUserAuthDetails);
+  const { getFeatureLimitValue, checkIfFeatureLimitReached } = useFeatureLimiter();
+  const [openPopup, setOpenPopup] = useState(false);
+
+  const isUpgradePopoverEnabled = useFeatureIsOn("show_upgrade_popovers");
+  const showPremiumPopovers = useMemo(
+    () => features.some((feat) => !(getFeatureLimitValue(feat) && !checkIfFeatureLimitReached(feat, "reached"))),
+    [features, getFeatureLimitValue, checkIfFeatureLimitReached]
+  );
+  const isBreachingLimit = features.some((feat) => checkIfFeatureLimitReached(feat, "reached"));
+
+  const hideUseForNowCTA = new Date() > new Date("2023-11-30");
+
+  useEffect(() => {
+    return () => {
+      setOpenPopup(false);
+    };
+  }, []);
+
+  return (
+    <>
+      {user?.details?.organization && !disabled && features ? (
+        <>
+          <RequestFeatureModal
+            isOpen={openPopup}
+            setOpenPopup={setOpenPopup}
+            organizationsData={user?.details?.organization}
+            onContinue={onContinue}
+            hasReachedLimit={isBreachingLimit}
+            source={source}
+          />
+          {React.Children.map(children, (child) => {
+            return React.cloneElement(child as React.ReactElement, {
+              onClick: () => {
+                if (showPremiumPopovers && isUpgradePopoverEnabled) setOpenPopup(true);
+                else onContinue();
+              },
+            });
+          })}
+        </>
+      ) : (
+        <Popconfirm
+          disabled={!showPremiumPopovers || !features || disabled || !isUpgradePopoverEnabled}
+          overlayClassName="premium-feature-popover"
+          autoAdjustOverflow
+          showArrow={false}
+          placement={popoverPlacement}
+          okText="See upgrade plans"
+          cancelText="Use for now"
+          onConfirm={() => {
+            trackUpgradeOptionClicked("see_upgrade_plans");
+            dispatch(actions.toggleActiveModal({ modalName: "pricingModal", newValue: true }));
+          }}
+          onCancel={() => {
+            if (!hideUseForNowCTA) {
+              trackUpgradeOptionClicked("use_for_free_now");
+              onContinue();
+            }
+          }}
+          cancelButtonProps={{ style: { display: hideUseForNowCTA ? "none" : "inline-flex" } }}
+          title={
+            <>
+              <Typography.Title level={4}>
+                {isBreachingLimit
+                  ? `${
+                      capitalize(getPrettyPlanNameFromId(user?.details?.planDetails?.planId)) || "Free"
```

Comment:
We do not need `getPrettyPlanNameFromId`. We can directly use `user.details.planDetails.planName`